cmd/bucket-replication.go

```diff
@@ -868,9 +868,12 @@ func NewReplicationPool(ctx context.Context, o ObjectLayer, opts replicationPool
 		replicaCh:       make(chan ReplicateObjectInfo, 100000),
 		replicaDeleteCh: make(chan DeletedObjectVersionInfo, 100000),
 		mrfReplicaCh:    make(chan ReplicateObjectInfo, 100000),
+		workerKillCh:    make(chan struct{}, opts.Workers),
```

Comment:
When killing, 
```
	for p.mrfWorkerSize > n {
		p.mrfWorkerSize--
		go func() { p.mrfWorkerKillCh <- struct{}{} }()
	}
```	
You can just close the channel since you aren't waiting for anything anyway.